vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -479,15 +476,16 @@ def check(self, _):
             self.collect_events()
 
         # Submit the number of VMs that are monitored
-        for resource in self.config.collected_resource_types:
-            # Explicitly do not attach any host to those metrics.
-            resource_count = len(self.infrastructure_cache.get_mors(resource))
-            self.gauge(
-                '{}.count'.format(MOR_TYPE_AS_STRING[resource]),
-                resource_count,
-                tags=self.config.base_tags,
-                hostname=None,
-            )
+        for resource_type in self.config.collected_resource_types:
+            for _, mor_props in self.infrastructure_cache.iter_all(resource_type):
+                # Explicitly do not attach any host to those metrics.
+                resource_tags = mor_props.get('tags', [])
+                self.count(
+                    '{}.count'.format(MOR_TYPE_AS_STRING[resource_type]),
+                    1,
+                    tags=self.config.base_tags + resource_tags,
+                    hostname=None,
+                )
```

Comment:
Could you add a test to cover those new metrics and behaviour? or include the case in an existing test.

Reply:
Of course